### PR TITLE
remove aws-sdk from ranger-extension

### PR DESCRIPTION
**Pull request by janjwerner-confluent**
Fixes # size blowup regression introduced in https://github.com/apache/druid/pull/15443

### Description

This PR removes the transitive dependency of ranger-plugins-audit to reduce the size of the compiled artifacts

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by LakshSingla** [comment]
Is it safe to exclude the dependency? Last I checked, the ranger did rely on AWS Cloudwatch for some things. Do we not use that functionality in Druid?

File: extensions-core/druid-ranger-security/pom.xml [hunk 1]
```diff
@@ -178,6 +178,10 @@
                     <groupId>org.apache.hadoop</groupId>
                     <artifactId>hadoop-common</artifactId>
                 </exclusion>
+                <exclusion>
+                    <groupId>com.amazonaws</groupId>
+                    <artifactId>aws-java-sdk-bundle</artifactId>
+                </exclusion>
```
Comment (xvrl) [inline extensions-core/druid-ranger-security/pom.xml:184]: it'd be nice to add some comments to explain how those are safe to exclude in our case and why we are excluding them.

Comment (janjwerner-confluent) [reply extensions-core/druid-ranger-security/pom.xml:185]: AWS-sdk-bundle is used only in CloudWatch audit [destination](https://github.com/apache/ranger/blob/master/agents-audit/src/main/java/org/apache/ranger/audit/destination/AmazonCloudWatchAuditDestination.java), and I believe that codepath is not touched by ranger extension for Druid, 
I will add  [aws-java-sdk-logs](https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-logs/1.12.669) package as explicit dependency of this module.

File: extensions-core/druid-ranger-security/pom.xml [hunk 2]
```diff
@@ -193,7 +198,13 @@
             </exclusions>
             <scope>compile</scope>
         </dependency>
-
+        <!-- This is a transitive dependency of ranger-plugins-audit
+             added here to replace bloated bundle sdk -->
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-logs</artifactId>
+            <version>1.12.669</version>
```
Comment (xvrl) [inline extensions-core/druid-ranger-security/pom.xml:206]: can we put a note next to the ranger plugin version to update the aws-sdk version when the ranger plugin gets updated?

**Review by xvrl — APPROVED** [review]

**Review by xvrl — APPROVED** [review]
LGTM once static checks / tests pass.

File: extensions-core/druid-ranger-security/pom.xml [hunk 3]
```diff
@@ -193,7 +193,14 @@
...
             <scope>compile</scope>
         </dependency>
-
+        <!-- This is a transitive dependency of ranger-plugins-audit
+             added here to replace bloated bundle sdk, remove this if /when
+             apache ranger replaces bundle-skd with logs-sdk  -->
+        <dependency>
+            <groupId>com.amazonaws</groupId>
+            <artifactId>aws-java-sdk-logs</artifactId>
+            <version>1.12.669</version>
```
Comment (xvrl) [inline extensions-core/druid-ranger-security/pom.xml:202]: nit, any reason we don't use ${aws.sdk.version} here and align the versions across the board?

**Comment by abhishekagarwal87** [comment]
why does the description says "azure-sdk"?

**Comment by janjwerner-confluent** [comment]
> why does the description says "azure-sdk"?

brain freeze I guess, fixed it :)

**Review by abhishekagarwal87 — APPROVED** [review]

File: extensions-core/druid-ranger-security/pom.xml (hunk 3 above)
Comment (janjwerner-confluent) [reply extensions-core/druid-ranger-security/pom.xml]: fixed

**Pull request merged**